00-programming-fundamentals/intro-to-oauth.md

```diff
@@ -0,0 +1,46 @@
+
+# Introduction to OAuth
+
+## Learning Goal
+You should be able to:
+*  Explain the OAuth workflow and how an OAuth service can provide authentication and information about a user.
+
+## Introduction
+
+OAuth is an authorization framework, meaning it allows a user to grant access to their user account in one service (the OAuth provider) to another (the client).  At Ada we will primarily use OAuth to use another service (Github) to provide secure authentication to our sites.  
+
+## OAuth Roles
+
+In an OAuth application the following entities play roles in the process.
+
+* The **User** is the person using the browser to access the site.
+* The **Client** is an application which wants access to the user's account.  
+* The **OAuth Provider** is the service providing OAuth access.  It maintains the following services.  
+  * The **Resource Server** is a server managed by the provider which hosts the user accounts.  
+  * The **Authorization Server** verifies the user's identity and provides access tokens to the client.
```

Comment:
Two suggestions:
- You should add something like "(our Rails app)" to the end of the Client line, and "(GitHub)" to the end of the OAuth Provider line.
- Since you don't ever talk about the Resource Server below, you could probably omit the distinction between the two sub-bullets here. Basically just treat the provider as a monolithic thing that keeps track of identities.